idl/github.com/uber/cadence/admin.thrift

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+namespace java com.uber.cadence.admin
+
+include "shared.thrift"
+
+exception AccessDeniedError {
+  1: required string message
+}
+
+/**
+* AdminService provides advanced APIs for debugging and analysis with admin privillege
+**/
+service AdminService {
+  /**
+  * InquiryWorkflowExecution returns information about the internal states of workflow execution.
+  **/
+  InquiryWorkflowExecutionResponse InquiryWorkflowExecution(1: shared.DescribeWorkflowExecutionRequest inquiryRequest)
+    throws (
+      1: shared.BadRequestError badRequestError,
+      2: shared.InternalServiceError internalServiceError,
+      3: shared.EntityNotExistsError entityNotExistError,
+      4: AccessDeniedError accessDeniedError,
```

Comment:
make this shared? access denied can be used for all other APIs in the future

Reply:
I feel like it will only be used in admin service in the near future, and our shared.thrift is already too big, so I put it here. I think it is never hard to move it to shard.thrift.

Reply:
i think we will have to eventually add access control, i.e. eventually throw AccessDeniedError to caller. @samarabbas see what is his preference

Reply:
yeah, I am open for it. Let's see our plan. If we are going to do some privilege checking for all frontend APIs, putting it in shared will make good sense. Let me ping him offline.